dev-tools/mage/target/unittest/unittest.go

```diff
@@ -51,6 +52,15 @@ func UnitTest() {
 	mg.SerialDeps(GoUnitTest, PythonUnitTest)
 }
 
+// GoFIPSOnlyUnitTest sets GODEBUG=fips140=only when running unit tests
+func GoFIPSOnlyUnitTest() error {
+	ctx := context.Background()
+	mg.SerialCtxDeps(ctx, goTestDeps...)
+
+	fipsArgs := devtools.DefaultGoFIPSOnlyTestArgs()
+	return devtools.GoTest(ctx, fipsArgs)
+}
```

Comment:
I chose to add a new mage target where the args have `GODEbuG=fips140=only` set instead of specifying it though a buildkite env var (as we do for `FIPS=true`) because I was not able to get it running locally, i.e,
```
$ FIPS=true GODEBUG=fips140=only mage goUnitTest
Error getting exe name: can't write data to hash: crypto/sha1: use of SHA-1 is not allowed in FIPS 140-only mode
```